BS440.py

```diff
@@ -53,35 +84,34 @@ def decodePerson(handle, values):
         retDict["activity"] = "normal"
     return retDict
 
-def sanitize_timestamp(timestamp):
-    retTS = 0
-    if timestamp + time_offset < sys.maxsize:
-        retTS = timestamp + time_offset
-    else:
-        retTS = timestamp
-
-    if timestamp >= sys.maxsize:
-        retTS = 0
-
-    return retTS
-
 
 def decodeWeight(handle, values):
     '''
     decodeWeight
-    Handle: 0x1b
-    Byte[0] = 0x1d
-    Returns:
-        valid (True, False)
-        weight (5,0 .. 180,0 kg)
-        timestamp (unix timestamp date and time of measurement)
-        person (1..9)
-        note: in python 2.7 to force results to be floats,
-        devide by float.
-        '''
+    Handle: 0x1b (Weight)             
+    Value:
+        Byte  Data                         Value/Return       Interpretation pattern
+         0    fixed byte (validity check)  [0x1d]             B (integer, length 1)
+         1    weight                       [5,0..180,0 kg]    H (integer, length 2)                     
+         2    weight                       
+         3    -pad byte-                                      x (pad byte)
+         4    -pad byte-                                      x (pad byte)   
+         5    timestamp                    Unix, date & time  I (integer, length 4)
+         6    timestamp              
+         7    timestamp                                 
+         8    timestamp
+         9    -pad byte-                                      x (pad byte) 
+        10    -pad byte-                                      x (pad byte) 
+        11    -pad byte-                                      x (pad byte) 
+        12    -pad byte-                                      x (pad byte) 
+        13    person                       [1..8]             B (integer, length 1)
+        --> Interpretation pattern:                           BHxxIxxxxB              
+    '''
     data = unpack('<BHxxIxxxxB', bytes(values[0:14]))
     retDict = {}
     retDict["valid"] = (data[0] == 0x1d)
+    # Weight is reported in g. Hence, divide by 100.0
```

Comment:
I'm not sure anymore where the information is from, but converting weight from `g` to `kg` would require division by `1000` :smile: . So Conversely I'd say the precision is actually 10g

Reply:
Good point. Did update the comment. Slipped my mind when I digged through the code.